distributed/tests/test_core.py

```diff
@@ -271,26 +271,24 @@ async def listen_on(cls, *args, **kwargs):
 
 
 async def check_rpc(listen_addr, rpc_addr=None, listen_args={}, connection_args={}):
-    server = Server({"ping": pingpong})
-    await server.listen(listen_addr, **listen_args)
-    if rpc_addr is None:
-        rpc_addr = server.address
-
-    async with rpc(rpc_addr, connection_args=connection_args) as remote:
-        response = await remote.ping()
-        assert response == b"pong"
-        assert remote.comms
+    async with Server({"ping": pingpong}) as server:
+        await server.listen(listen_addr, **listen_args)
+        if rpc_addr is None:
+            rpc_addr = server.address
 
-        response = await remote.ping(close=True)
-        assert response == b"pong"
-        response = await remote.ping()
-        assert response == b"pong"
+        async with rpc(rpc_addr, connection_args=connection_args) as remote:
+            response = await remote.ping()
+            assert response == b"pong"
+            assert remote.comms
 
-    assert not remote.comms
-    assert remote.status == Status.closed
+            response = await remote.ping(close=True)
+            assert response == b"pong"
+            response = await remote.ping()
+            assert response == b"pong"
 
-    server.stop()
-    await asyncio.sleep(0)
+        assert not remote.comms
+        assert remote.status == Status.closed
+        await asyncio.sleep(0)
```

Comment:
is this `sleep(0)` needed anymore?

Reply:
I don't know.  I don't really want to get into fixing these tests though (low priority I think).  I consider that out of scope for this PR.  

https://matthewrocklin.com/blog/work/2022/03/26/fast-review